src/lib/features/events/event-store.ts

```diff
@@ -147,6 +157,22 @@ class EventStore implements IEventStore {
         }
     }
 
+    async searchEventsCount(
+        params: IEventSearchParams,
+        queryParams: IQueryParam[],
+    ): Promise<number> {
+        const query = this.buildSearchQuery(params, queryParams);
+        const count = await query.count().first();
+        if (!count) {
+            return 0;
+        }
+        if (typeof count.count === 'string') {
+            return Number.parseInt(count.count, 10);
+        } else {
+            return count.count;
+        }
```

Comment:
Why is this necessary? Does the db sometimes return the count as a string and other times as a number? Maybe it's the TS type that's doing it?